src/Abp.AutoMapper/AutoMapper/AutoMapperConfigurationExtensions.cs

```diff
@@ -13,5 +16,104 @@ public static void CreateAutoAttributeMaps(this IMapperConfigurationExpression c
                 autoMapAttribute.CreateMap(configuration, type);
             }
         }
+
+        public static void CreateAutoAttributeMaps(this IMapperConfigurationExpression configuration, Type type, Type[] targetTypes, MemberList memberList)
+        {
+
+            //Get all the properties in the source that have the AutoMapKeyAttribute
+            var sourceKeysPropertyInfo = type.GetProperties()
+                                             .Where(w => w.GetCustomAttribute<AutoMapKeyAttribute>() != null)
+                                             .Select(s => s).ToList();
+
+            foreach (var targetType in targetTypes)
+            {
+                if (!sourceKeysPropertyInfo.Any())
+                {
+                    configuration.CreateMap(type, targetType, memberList);
+                    continue;
+                }
+                BinaryExpression equalityComparer = null;
+
+                //In a lambda expression represent the source exemple : (source) => ...
+                ParameterExpression sourceParameterExpression = Expression.Parameter(type, "source");
+                //In a lambda expression represent the target exemple : (target) => ...
+                ParameterExpression targetParameterExpression = Expression.Parameter(targetType, "target");
+
+
+                //We could use multiple AutoMapKey to compare the determine equality
+                foreach (PropertyInfo propertyInfo in sourceKeysPropertyInfo)
+                {
+                    //In a lambda expression represent a specfic property of a parameter exemple : (source) => source.Id
+                    MemberExpression sourcePropertyExpression = Expression.Property(sourceParameterExpression, propertyInfo);
+
+                    //Find the target a property with the same name to compare with
+                    //Exemple if we have in source the attribut AutoMapKey on the Property Id we want to get Id in the target to compare agaisnt
+                    var targetPropertyInfo = targetType.GetProperty(sourcePropertyExpression.Member.Name);
+
+                    //It happen if the property with AutoMapKeyAttribute does not exist in target
+                    //Maybe we should throw or maybe it's ok the compare only the existing match
```

Comment:
I think it is ok to continue since we don't use AutoMapper's AssertConfigurationIsValid.